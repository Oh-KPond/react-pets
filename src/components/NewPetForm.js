import React, { Component } from 'react';
import PropTypes from 'prop-types';

class NewPetForm extends Component {
  constructor() {
    super();

    this.state = {
      name: '',
      age: 0,
      breed: '',
      about: ''
    };
  }

  onFieldChange = (event) => {
    const fieldName = event.target.name;
    const fieldValue = event.target.value;
    const updateState = {};
    updateState[fieldName] = fieldValue;
    this.setState(updateState);
  }


  render() {
    return (
      <form>
        <div>
          <label htmlFor="name">Name: </label>
          <input name="name" value={this.state.className}
          onChange={this.onFieldChange} type="text"
          />

          <label htmlFor="age">Age: </label>
          <input name="age" value={this.state.age}
          onChange={this.onFieldChange} type="number"
          />

          <label htmlFor="breed">Breed: </label>
          <input name="breed" value={this.state.breed}
          onChange={this.onFieldChange} type="text"
          />

          <label htmlFor="about">About: </label>
          <textarea name="about" value={this.state.about}
          onChange={this.onFieldChange}
          />

        </div>
      </form>
    )
  }
}

export default NewPetForm;
