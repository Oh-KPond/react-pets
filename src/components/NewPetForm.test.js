import React from 'react';
import { mount } from 'enzyme';

import NewPetForm from './NewPetForm';

describe('NewPetForm', () => {
  test('that it matches an existing snapshot', () => {
  // Mount the component in the DOM
  // Deep Rendering

  // Given
  const wrapper = mount(<NewPetForm />);
  // When and Then
  expect(wrapper).toMatchSnapshot();
  // Remove component from the DOM
  wrapper.unmount();
  });
});
