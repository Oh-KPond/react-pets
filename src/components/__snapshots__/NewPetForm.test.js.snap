// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPetForm that it matches an existing snapshot 1`] = `
<NewPetForm
  addPetCallback={[Function]}
>
  <form
    onSubmit={[Function]}
  >
    <div>
      <label
        htmlFor="name"
      >
        Name: 
      </label>
      <input
        name="name"
        onChange={[Function]}
        type="text"
        value=""
      />
      <label
        htmlFor="age"
      >
        Age: 
      </label>
      <input
        name="age"
        onChange={[Function]}
        type="number"
        value={0}
      />
      <label
        htmlFor="breed"
      >
        Breed: 
      </label>
      <input
        name="breed"
        onChange={[Function]}
        type="text"
        value=""
      />
      <label
        htmlFor="about"
      >
        About: 
      </label>
      <textarea
        name="about"
        onChange={[Function]}
        value=""
      />
      <input
        type="submit"
        value="add Pet"
      />
    </div>
  </form>
</NewPetForm>
`;
